Added new variable RMW_CONNEXT_DISCOVERY_TOPICS_PUBLISH_MODE (#240)
rticommunity/rmw_connextdds: README.md

### RMW_CONNEXT_USER_TOPICS_PUBLISH_MODE

`rmw_connextdds` will always set `DDS_DataWriterQos::publish_mode::kind` of any
user DataWriter it creates to the default value of Connext (Synchronous publish
mode) or to the value of a provided XML file.

To override this behavior, you can set this variable to any of these values:

- *synchronous*: Set publish mode of every user DataWriter to Synchronous.
- *asynchronous*: Set publish mode of every user DataWriter to Asynchronous.
- *auto*: Connext will choose the publish mode according to the QoS XML
  configuration. If no XML file is used, Connext will use the default publish
  mode (Synchronous).

This variable is not used by `rmw_connextddsmicro`, since it doesn't
automatically override `DDS_DataWriterQos::publish_mode::kind`.

> ### RMW_CONNEXT_DISCOVERY_TOPICS_PUBLISH_MODE

Reviewer: Do we only care about the discovery built in topics or should this apply to all built in topics?

I'm thinking of 
`participant_configuration_writer_publish_mode`, `service_request_writer_publish_mode`, and `secure_volatile_writer_publish_mode`

If we really ONLY care about discovery topics, then `participant_configuration_writer_publish_mode` should still apply as it is for participant discovery (SPDP2)

Author: Just discovery topics. I will add `participant_configuration_writer_publish_mode` too.

This variable is not used by `rmw_connextddsmicro`, since it doesn't
automatically override `DDS_DataWriterQos::publish_mode::kind`.

> ### RMW_CONNEXT_DISCOVERY_TOPICS_PUBLISH_MODE

Author: @lloadholt can you review the changes in the README?

`rmw_connextdds` will always set
`DDS_DiscoveryConfigQosPolicy::publication_writer_publish_mode::kind` and
`DDS_DiscoveryConfigQosPolicy::subscription_writer_publish_mode::kind` of any
built-in Publication and Subscription Data Writer to the default value of
Connext (Synchronous publish mode) or to the value of a provided XML file.

To override this behavior, you can set this variable to any of these values:

- *synchronous*: Set publish mode of every built-in Publication and Subscription
  Data Writer to Synchronous.

> - *asynchronous*: Set publish mode of every built-in Publication and

Reviewer 2: ```suggestion
- *asynchronous*: Set publish mode of every built-in publication and
```

>   Subscription Data Writer to Asynchronous.

Reviewer 2: ```suggestion
  subscription DataWriter to Asynchronous.
```

> - *auto*: Connext will choose the publish mode according to the QoS XML

Reviewer 2: ```suggestion
- *auto*: Connext chooses the publish mode according to the QoS XML
```

  configuration. If no XML file is used, Connext will use the default publish
  mode (Synchronous).

> This variable is not used by `rmw_connextddsmicro`, since it doesn't

Reviewer 2: ```suggestion
This variable is not used by `rmw_connextddsmicro` because it doesn't
```
